pkg/api/controllers/volume.go

```diff
@@ -239,6 +239,36 @@ func (v *VolumePortal) ExtendVolume() {
 		return
 	}
 
+	if volume.Status != model.VolumeAvailable && volume.Status != model.VolumeErrorExtending {
+		errMsg := "the status of the volume to be extended must be available or error extending!"
+		v.ErrorHandle(model.ErrorBadRequest, errMsg)
+		return
+	}
+
+	if extendRequestBody.NewSize <= volume.Size {
+		errMsg := fmt.Sprintf("new size for extend must be greater than current size."+
+			"(current: %d GB, extended: %d GB).", volume.Size, extendRequestBody.NewSize)
+		v.ErrorHandle(model.ErrorBadRequest, errMsg)
+		return
+	}
+
+	pool, err := db.C.GetPool(ctx, volume.PoolId)
+	if err != nil {
+		errMsg := fmt.Sprintf("pool %s not found: %s", id, err.Error())
+		v.ErrorHandle(model.ErrorNotFound, errMsg)
+		return
+	}
+	if pool.FreeCapacity < extendRequestBody.NewSize-volume.Size {
+		errMsg := fmt.Sprintf("pool:%s capacity:%v is not enough", pool.Name, pool.FreeCapacity)
+		v.ErrorHandle(model.ErrorBadRequest, errMsg)
+		return
+	}
+	if pool.Status != model.PoolAvailable {
+		errMsg := fmt.Sprintf("pool:%s with status:%s can not support extending volume", pool.Name, pool.Status)
```

Comment:
Should the message be 
```suggestion
		errMsg := fmt.Sprintf("pool:%s has status:%s. Can not  extend volume", pool.Name, pool.Status)
```